File: src/database/low.ts
import fs from 'fs'
import path from 'path'
import { LowResponse } from './types'



function getData(): LowResponse {
    try {
        const raw = fs.readFileSync(
            path.join(__dirname, 'data.json'), 
            'utf8'
        )
        try { return { success: true, message: JSON.parse(raw) } }
        catch { return { success: false, message: 'Error parsing JSON' } }
    } catch { return { success: false, message: 'Error getting data' } }
}
function setData(data: Object) {
    try {
        const parsed = JSON.stringify(data)
        try {
            fs.writeFileSync(
                path.join(__dirname, 'data.json'),
                parsed,
                { encoding: 'utf-8' }
            )
            return { success: true, message: 'Data saved' }
        } catch { return { success: false, message: 'Error saving data' } }
    } catch { return { success: false, message: 'Error parsing data' } }
}


export default {
    getData,
    setData
}

File: src/database/low.rs
use std::fs::{read_to_string, write};
use super::Error;
use json::{parse, stringify, JsonValue};

pub fn get_data() -> Result<JsonValue, Error> {
    let data = read_to_string("data.json");
    match data {
        Err(_) => Err(Error::FailedToOpen),
        Ok(safe) => parser(&safe),
    }
}

fn parser(safe: &str) -> Result<JsonValue, Error> {
    match parse(safe) {
        Err(_) => Err(Error::FailedToParse),
        Ok(safeparsed) => Ok(safeparsed),
    }
}

pub fn set_data(obj: JsonValue) -> Result<JsonValue, Error> {
    let data = stringify(obj);
    let response = write("data.json", data.as_bytes());
    match response {
        Err(_) => Err(Error::FailedToSave),
        Ok(_) => Ok(JsonValue::Null),
    }
}
